modules/kernel/src/main/scala/trace4cats/kernel/headers/GoogleCloudTraceToHeaders.scala

```diff
@@ -28,26 +28,31 @@ private[trace4cats] object GoogleCloudTraceToHeaders {
   // from https://cloud.google.com/trace/docs/setup
   val headerPattern =
     """(?xi)
-      |([0-9a-f]+) # trace ID
+      |([^\/]+)    # trace ID
       |\/
-      |(\d+)       # span ID (unsigned decimal)
+      |([^;]+)     # span ID (unsigned decimal)
+      |(?:
       |;
-      |o=(0|1)     # trace enabled flag
+      |o=(.*)      # trace enabled flag
+      |)?
       |""".stripMargin.r
 
   def parse(header: String): Either[Throwable, SpanContext] = header match {
     case headerPattern(traceId, spanId, enabled) =>
       for {
         traceId <- Either.fromOption(TraceId.fromHexString(traceId), new Exception("invalid trace ID"))
         spanId <- Either.fromOption(
-          SpanId.fromHexString("%016x".format(BigInt(spanId))),
+          Either.catchNonFatal(BigInt(spanId)).toOption.flatMap(bi => SpanId.fromHexString("%016x".format(bi))),
           new Exception("invalid span ID")
```

Comment:
The parse method supports surfacing an exception on the left of an either so I'd suggest doing this instead:

`Either.catchNonFatal(BigInt(spanId)).flatMap(bi => Either.fromOption(SpanId.fromHexString("%016x".format(bi)), new Exception("invalid span ID")))`

Reply:
Done :+1: